PostgreSQL native bulk copy

MaceWindu opened the pull request:
Continuation of #935 

Remaining tasks:
- [x] make types test work (add non-null values and fix all discovered issues)
- [x] update BulkCopy wiki article with PostgreSQL changes: https://github.com/linq2db/linq2db/wiki/Bulk-Copy
- [ ] update BulkCopy wiki article with documentation of more BulkOptions properties
- [x] retest all databases as some changes done to default mappings
- [x] fix code formatting :-/

sdanyliv commented on Source/LinqToDB/DataProvider/PostgreSQL/PostgreSQLDataProvider.cs:
```diff
@@ -291,13 +393,149 @@ public override CommandBehavior GetCommandBehavior(CommandBehavior commandBehavi
+					dbType = "bit varying";
+					break;
+			}
+
```
Not sure but regular expression looks more suitable here, especially with code below.

MaceWindu replied:
I would prefere to not use regexps in library code :)

jogibear9988 replied:
I think a regex will be less performant

MaceWindu commented:
This one is ready for review.

Short overview of changes:
- native bulk copy for postgresql with test for all types (also should work with npgsql 4.0 when it will be released, as it contains breaking changes to this functionality)
- updated AllTypes test table names to reduce number of postgresql-specific test mappings
- added missing and updated existing postgresql types support (different versions of provider have different sets of types and some of them were missing/outdated)
- updated type name generation for some postgresql types to generate proper type name
